modules/rostests/apitests/shell32/ShellExecuteEx.cpp

```diff
@@ -81,8 +155,67 @@ TestShellExecuteEx(const WCHAR* Name, BOOL ExpectedResult)
     }
 }
 
+static
+VOID
+TestShellExecuteEx1(const WCHAR* Name, const WCHAR* Params)
+{
+    SHELLEXECUTEINFOW ShellExecInfo;
+    BOOL Result;
+    CHAR OSVendor[100] = {0};
+    INT iResult;
+    UINT_PTR retval = SE_ERR_NOASSOC;
+
+    ZeroMemory(&ShellExecInfo, sizeof(ShellExecInfo));
+    ShellExecInfo.cbSize = sizeof(ShellExecInfo);
+    ShellExecInfo.fMask = SEE_MASK_NOCLOSEPROCESS | SEE_MASK_FLAG_NO_UI;
+    ShellExecInfo.hwnd = NULL;
+    ShellExecInfo.nShow = SW_SHOWNORMAL;
+    ShellExecInfo.lpFile = Name;
+    ShellExecInfo.lpParameters = Params;
+    ShellExecInfo.lpDirectory = NULL;
+    ShellExecInfo.lpVerb = L"open";
+
+    iResult = GetOSVendor(OSVendor, sizeof(OSVendor));
+    printf("OSVendor is '%s'.\n", OSVendor);
+
+    if(strcmp(OSVendor, "Microsoft") == 0)
+        skip("Test does not work on MS Windows.\n");
+
+    if(iResult == FALSE)
+        skip("Unable to determine Operating System.\n");
+
+    Result = ShellExecuteExW(&ShellExecInfo);
+
+    ok(Result, "ShellExecuteEx lpFile %s failed. Error: %lu\n",
+                wine_dbgstr_w(Name), GetLastError());
+
+    if(strcmp(OSVendor, "Microsoft") == 0)
+        skip("Test does not work on MS Windows.\n");
+
+    if(iResult == FALSE)
+        skip("Unable to determine Operating System.\n");
```

Comment:
These 2 checks duplicate previous code...

Reply:
@SergeGautherie, This is true, but for a different type of test. The first messages apply to whether we could read the OS Vendor. The second set of messages apply to the results of ShellExecuteExW and why it did not work. Therefore, I contend that this does its job correctly.

Reply:
As far as I can tell `iResult` and `OSVendor` have nothing to do with `ShellExecuteExW()`, so I see no purpose in checking/"reporting" the same things again.